### PR TITLE
Post-cmake cleanup and decommissioning

### LINKED ISSUE #272: Post-cmake cleanup and decommissioning
Perform cleanup after cmake build updates done in PR #264 and decommission now-unneeded files.

- [x] update build_all.sh - remove unneeded build scripts
- [x] remove unneeded build scripts that are now combined into build_workflow_utils.sh
- [x] remove unneeded modulefiles under /modulefiles
- [x] remove unneeded builds from partial_build.sh and fv3gfs_build.cfg

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
Cleanup after cmake build updates done in PR #264 and decommissioning of now-unneeded files:

- updated build_all.sh; removed unneeded build scripts
- removed unneeded build scripts that were combined into build_workflow_utils.sh
- removed unneeded modulefiles under /modulefiles
- removed unneeded builds from partial_build.sh and fv3gfs_build.cfg

Ran 2.5 cycles on WCOSS-Dell to confirm cleanup did not break anything. Test was successful.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -81,8 +81,10 @@ sorc/recentersigp.fd
 sorc/shave.fd
 sorc/wafs_awc_wafavn.fd
 sorc/wafs_blending.fd
+sorc/wafs_blending_0p25.fd
```
Comment (WalterKolczynski-NOAA) [inline .gitignore:84]: Why is WAFS getting additional directories?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (KateFriedman-NOAA) [reply .gitignore:84]: In prior commit to develop (bringing in v16.0.7 changes from NCO) it included adding two missing sorc folder symlinks to link_fv3gfs.sh (which Yali Mao and SPA Jen noticed). They needed to be added to the .gitignore file but were overlooked in prior commit and I noticed in a fresh clone for this task. So the sorc folders already existed, just needed to ignore them.

Comment (WalterKolczynski-NOAA) [reply .gitignore:84]: We should talk to WAFS about not having a separate resolution-dependent source directory for this in the future.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (KateFriedman-NOAA) [reply .gitignore:84]: Definitely, we can open an issue with Yali for that. The specific sorc folder symlinks will most likely go away in the future anyway...SPA Jen indicated during chats about this that they don't see them as needed anymore so we can reevaluate them for v17.

**Comment by KateFriedman-NOAA** [comment]
Close #272

**Pull request merged**